sense-android-library-test/src/nl/sense_os/senseservice/TestDSEDataConsumer.java

```diff
@@ -0,0 +1,117 @@
+ */
+public class TestDSEDataConsumer extends AndroidTestCase {
+    // The default source for the Android sensing library
+    /**
+     * The sensor names for which dummy sensors are created
+     * @return The list with the names of the available dummy sensors
+     */
+    public ArrayList<String> getSensorNames() {
+        ArrayList<String> sensorNames = new ArrayList();
+        sensorNames.add(SensorData.SensorNames.POSITION);
+        sensorNames.add(SensorData.SensorNames.NOISE);
+        sensorNames.add(SensorData.SensorNames.TIME_ZONE);
+        sensorNames.add(SensorData.SensorNames.ACCELEROMETER);
+        sensorNames.add(SensorData.SensorNames.BATTERY);
+        sensorNames.add(SensorData.SensorNames.CALL);
+        sensorNames.add(SensorData.SensorNames.LIGHT);
+        sensorNames.add(SensorData.SensorNames.PROXIMITY);
+        sensorNames.add(SensorData.SensorNames.SCREEN);
+        sensorNames.add(SensorData.SensorNames.WIFI_SCAN);
+        sensorNames.add(SensorData.SensorNames.APP_INFO);
+        return sensorNames;
+    }
+
+    @Override
+    protected void setUp() throws Exception {
+        SenseServiceUtils.createAccountAndLoginService(getContext());
+        super.setUp();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        SenseServiceUtils.logout(getContext());
+        super.tearDown();
+    }
+
+    /**
+     *  Test the storage of sensor data via the DSEDataConsumer
+     */
+    public void testDSEDataConsumer() throws InterruptedException, ExecutionException, TimeoutException, JSONException, SchemaException, DatabaseHandlerException, SensorProfileException, SensorException {
+        // Get the DSE
+        DataStorageEngine dataStorageEngine = DataStorageEngine.getInstance(getContext());
+        // Wait until the DSE is ready
+        dataStorageEngine.onReady().get(60, TimeUnit.SECONDS);
+        // send the sensor data
+        SensorUtils sensorUtils = SensorUtils.getInstance();
+        sensorUtils.accelerometerSensor.sendDummyData();
+        sensorUtils.appInfo.sendDummyData();
+        sensorUtils.batterySensor.sendDummyData();
+        sensorUtils.callSensor.sendDummyData();
+        sensorUtils.lightSensor.sendDummyData();
+        sensorUtils.noiseSensor.sendDummyData();
+        sensorUtils.positionSensor.sendDummyData();
+        sensorUtils.proximitySensor.sendDummyData();
+        sensorUtils.screenSensor.sendDummyData();
+        sensorUtils.timeZoneSensor.sendDummyData();
+        sensorUtils.wifiScanSensor.sendDummyData();
+        QueryOptions queryOptions = new QueryOptions();
+        // check if it's stored
+        for(String sensorName : getSensorNames()){
+            Sensor sensor = dataStorageEngine.getSensor(SensorData.SourceNames.SENSE_ANDROID, sensorName);
+            SensorOptions sensorOptions = sensor.getOptions();
+            assertTrue("Upload not enabled for this sensor", sensorOptions.isUploadEnabled());
```

Comment:
@orhan89 The default SensorOption is upload enabled for all these sensors, which should  automatically be created in the DSEDataConsumer.